apps/screloc/main.cpp

```diff
@@ -496,9 +540,31 @@ class Reloc : public Client::Application {
 			for ( size_t i = 0; i < org->arrivalCount(); ++i ) {
 				Arrival *ar = org->arrival(i);
 				PickPtr pick = _cache.get<Pick>(ar->pickID());
-				if ( !pick ) continue;
+				if ( !pick ) {
+					continue;
+				}
 
-				if ( !_useWeight ) {
+				// filter streams by whitelist
+				bool unusedStream = false;
+				string streamID = pick->waveformID().networkCode()
+				                  + "." + pick->waveformID().stationCode()
+				                  + "." + pick->waveformID().locationCode()
+				                  + "." + pick->waveformID().channelCode();
+				vector<string> toks;
+				for ( const auto &stream : _pickStreamsUnused ) {
+					if ( Core::wildcmp(stream, streamID) ) {
+						SEISCOMP_DEBUG("  + ignoring pick %s on stream %s",
+									   pick->publicID().c_str(), streamID);
+						ar->setWeight(0.0);
+						ar->setTimeUsed(false);
+						ar->setBackazimuthUsed(false);
+						ar->setHorizontalSlownessUsed(false);
+						unusedStream = true;
+						break;
+					}
+				}
+
+				if ( !_useWeight && !unusedStream ) {
```

Comment:
double negation `! unusedStream` is difficult to read ("if the stream is not unused"). You could  call the variable `ignoredStream` or reverse the logic and call the variable `usedStream`

Reply:
using positive clause